e2e/cypress/integration/bot_accounts/crud_spec.js

```diff
@@ -0,0 +1,278 @@
+        });
+    });
+
+    it('MM-T1842 Change BOT role', () => {
+        // # Visit the integrations
+        cy.visit(`/${newTeam.name}/integrations/bots`);
+
+        // * Check that the previously created bot is listed
+        cy.findByText(`Test Bot (@${botName})`).then((el) => {
+            // # Make sure it's on the screen
+            cy.wrap(el[0].parentElement.parentElement).scrollIntoView();
+
+            // # Click the edit button
+            cy.wrap(el[0].parentElement.parentElement).findByText('Edit').should('be.visible').click();
+
+            // * Validate redirect to edit screen
+            cy.url().
+                should('include', `/${newTeam.name}/integrations/bots/edit`);
+
+            // # Select sysadmin
+            cy.get('select').select('System Admin');
+
+            // * Validate that permissions are set and read only
+            cy.get('#postChannels').should('be.checked').should('be.disabled');
+            cy.get('#postAll').should('be.checked').should('be.disabled');
+
+            // # Update the bot
+            cy.get('#saveBot').click();
+
+            // * Validate that bot saved correctly
+            cy.url().should('include', `/${newTeam.name}/integrations/bots`);
+        });
+    });
+
+    function createBotInteractive(username = `bot-${getRandomId()}`) {
+        // # Visit the integrations
+        cy.visit(`/${newTeam.name}/integrations/bots`);
+
+        // # Click add bot
+        cy.get('#addBotAccount').click();
+
+        // # fill+submit form
+        cy.get('#username').type(username);
+        cy.get('#displayName').type('Test Bot');
+        cy.get('#saveBot').click();
+
+        // * verify confirmation page
+        cy.url().
+            should('include', `/${newTeam.name}/integrations/confirm`).
+            should('match', /token=[a-zA-Z0-9]{26}/);
+
+        // * verify confirmation form/token
+        cy.get('div.backstage-form').
+            should('include.text', 'Setup Successful').
+            should((confirmation) => {
+                expect(confirmation.text()).to.match(/Token: [a-zA-Z0-9]{26}/);
+            });
+
+        return cy.get('div.backstage-form').invoke('text');
+    }
+
+    it('MM-T1843 ID along with actual token is created', () => {
+        // # Create the bot and validate token is visible
+        createBotInteractive();
+        cy.get('#doneButton').click();
+    });
+
+    it('MM-T1844 Token is hidden whe you return to the page but ID is still visible', () => {
+        // # Create the bot and validate token is visible
+
+        const botUsername = `bot-${getRandomId()}`;
+
+        createBotInteractive(botUsername).then((text) => {
+            // # Close the Add dialog
+            cy.get('#doneButton').click();
+
+            // * Check that the previously created bot is listed
+            cy.findByText(`Test Bot (@${botUsername})`).then((el) => {
+                cy.wrap(el[0].parentElement.parentElement).scrollIntoView();
+
```

Comment:
I think the test is to verify that when the user leaves the bot backstage, goes back the normal MM channel view, then goes back into the bot page from main menu > integrations > bot, the personal access token is now hidden. Wondering if those navigation to channel and back to bot page is missing or if I understood it differently :)

Reply:
@prapti - I don't think the test case required going back to the channel, unless I misread it somehow? :thinking:

Reply:
@reflog, sorry for bugging again on this one. I'm looking at the test case and step #7 & #8: 
"7. Leave the Bots backstage UI
8. Return to the Bots backstage UI and click edit on the bit create above"
I thought "leave the bots backstage" meant exit out of it which will take the user to the channels; is that not the case? Wondering if @DHaussermann would know more about it?
So sorry, this one is lingering here, but the rest look great.

Reply:
@prapti no, there is need to go back to any channel on the user side. Sorry the step was overly specific which caused confusion. We just need to see the list of bots and ensure that the newly created token in not showing in the UI.

Reply:
@DHaussermann, "no, no need to go back", or "yes, need to go back"? Sorry, I keep coming back to this. But I think it's good to be approved and merged -- we can always come back to it later if we're missing steps since it won't be a significant effort as opposed to what's covered in this PR which has way more value right now :)